aspnetcore/blazor/test.md

```diff
@@ -0,0 +1,137 @@
+---
+title: Test components in ASP.NET Core Blazor
+author: guardrex
+description: Learn how to test componments in Blazor apps.
+monikerRange: '>= aspnetcore-3.1'
+ms.author: riande
+ms.custom: mvc
+ms.date: 08/10/2020
+no-loc: [cookie, Cookie, Blazor, "Blazor Server", "Blazor WebAssembly", "Identity", "Let's Encrypt", Razor, SignalR]
+uid: blazor/test
+---
+# Test components in ASP.NET Core Blazor
+
+[Egil Hansen](https://egilhansen.com/)
+
+Testing is an important aspect of building stable and maintainable software.
+
+To test a Blazor component, the *Component Under Test* (CUT) is:
+
+* Rendered with relevant input for the test.
+* Depending on the type of test performed, possibly subject to interaction or modification. For example, event handlers can be triggered, such as an `onclick` event for a button.
+* Inspected for expected values.
+
+## Test approaches
+
+Two common approaches for testing Blazor components are End-to-end (E2E) testing and unit testing:
```

Comment:
Should End-to-end really be capitalized?

Reply:
no ... I think I made a mistake there. I might have had a draft with "End-to-End" as a proper noun, but it's not really. I'll fix it.